Reaktoro/Core/Phases.hpp

```diff
@@ -1,387 +1,433 @@
+
+    /// Construct a LiquidPhase object with given element symbols excluding the species with provided tags.
+    explicit LiquidPhase(const Speciate& elements, const Exclude& withtags) { initialize(); };
+
+    /// Construct a LiquidPhase object excluding the species with provided tags.
+    explicit LiquidPhase(const Exclude& withtags) { initialize(); };
+
+    /// Initialize the default attributes of this LiquidPhase object.
+    auto initialize() -> void
+    {
+        setName("LiquidPhase");
+        setStateOfMatter(StateOfMatter::Liquid);
+        setAggregateState(AggregateState::Liquid);
+        setActivityModel(ActivityModelIdealSolution());
+        setIdealActivityModel(ActivityModelIdealSolution());
+    }
+};
+
+/// The class used to configure a solid solution phase.
+class SolidPhase : public GenericPhase
+{
+public:
+    /// Construct a SolidPhase object with given species names.
+    explicit SolidPhase(const StringList& species) : GenericPhase(species) { initialize(); }
+
+    /// Initialize the default attributes of this SolidPhase object.
+    auto initialize() -> void
+    {
+        String phasename;
+        auto i = 0;
+        for(auto&& name : species())
+            phasename += (i++ == 0) ? name : "-" + name;
+
+        setName("SolidPhase");
+        setStateOfMatter(StateOfMatter::Solid);
+        setAggregateState(AggregateState::Solid);
+        setActivityModel(ActivityModelIdealSolution());
+        setIdealActivityModel(ActivityModelIdealSolution());
+    }
+};
+
+/// The class used to configure a pure mineral phase.
+class MineralPhase : public GenericPhase
+{
+public:
+    /// Construct a default MineralPhase object.
+    explicit MineralPhase(String mineral) : GenericPhase(mineral) { initialize(); }
+
+    /// Initialize the default attributes of this MineralPhase object.
+    auto initialize() -> void
+    {
+        setName(species().front());
+        setStateOfMatter(StateOfMatter::Solid);
+        setAggregateState(AggregateState::Solid);
+        setActivityModel(ActivityModelIdealSolution());
+        setIdealActivityModel(ActivityModelIdealSolution());
+    }
+};
+
+/// The class used to configure automatic selection of pure mineral phases.
+class MineralPhases : public GenericPhasesGenerator
+{
+public:
+    /// Construct a default MineralPhases object.
+    MineralPhases() : GenericPhasesGenerator() { initialize(); }
+
+    /// Construct a MineralPhases object with given species names.
+    explicit MineralPhases(const StringList& species) : GenericPhasesGenerator(species) { initialize(); }
+
+    /// Construct a MineralPhases object with given element symbols.
+    explicit MineralPhases(const Speciate& elements) : GenericPhasesGenerator(elements) { initialize(); }
+
+    /// Construct a MineralPhases object with given element symbols excluding the species with provided tags
+    explicit MineralPhases(const Speciate& elements, const Exclude& withtags) { initialize(); };
```

Comment:
`GenericPhasesGenerator` should also have a constructor that accepts an `Exclude` object (similar to `GenericPhase`). This class, `GenericPhasesGenerator`, is used to create one or more phases with single species only. So, we want to avoid creating phases with species that have one or more "forbidden" tags. For example, `MineralPhases(speciate(...), exclude("carbonate"))`, will prevent the creation of mineral phases from carbonate minerals.

Reply:
Added